NTUEELightDance/LightDance-Editor
# EDITOR-#500 [BLENDER] Waveform

### What is this PR for?
Waveform in timeline and dopesheet, combined with EDITOR-#459

### What type of PR is it?
[Feature]

### Todos


### What is the Github issue?
"resolve #500 "

### How should this be tested?
The code will look for waveform file in the same folder as blender file, and if no file named 'waveform.png' exists in the folder, it will fetch waveform from file-server.
Open file-server on port 8081, and make sure that a waveform file already existed in file-server.
A waveform file with size of about 60000*400 is recommended. If no such file exists, you can use the waveform generator in utils. (Waveforms that are too large may cause distortion when rendering)
Click the "ON" button in the side menu, then the waveform should appear shortly.
The size of waveform can be adjusted by the 'height offset' and 'end frame' values in the side menu.


### Screenshots (if appropriate)

### Questions

* Do the license files need updating? Yes/No
* Are there breaking changes for older versions? Yes/No
* Does this need new documentation? Yes/No
